specification/impact/Impact.Management/workloadImpacts.tsp

```diff
@@ -0,0 +1,92 @@
+import "@typespec/openapi";
+import "@typespec/rest";
+
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+namespace Microsoft.Impact;
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using OpenAPI;
+using Azure.ResourceManager;
+
+@doc("Workload Impact properties")
+@subscriptionResource
+model WorkloadImpact is ProxyResource<WorkloadImpactProperties> {
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-:]*[a-zA-Z0-9]{3,120}$")
+  @key("workloadImpactName")
+  @segment("workloadImpacts")
+  @doc("workloadImpact resource ")
+  @visibility("Read")
+  @path
+  name: string;
+}
+
+@doc("Workload impact properties")
+model WorkloadImpactProperties is ResourceProperties {
+  @doc("Time at which impact was observed ")
+  startDateTime: utcDateTime;
+
+  @doc("Time at which impact has ended ")
+  endDateTime?: utcDateTime;
+
+  @doc("Azure resource id of the impacted resource")
+  impactedResourceId: string;
+
+  @doc("Unique ID of the impact (UUID)")
+  @visibility("read")
+  @pattern("[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}")
+  impactUniqueId?: string;
+
+  @doc("Time at which impact is reported")
+  @visibility("read")
+  reportedTimeUtc?: utcDateTime;
+
+  @doc("Category of the impact,  details can found from /impactCategories API")
+  impactCategory: string;
+
+  @doc("A detailed description of the impact")
+  impactDescription?: string;
+
+  @doc("The ARM correlation ids, this is important field for control plane related impacts")
+  @extension("x-ms-identifiers", [])
+  armCorrelationIds?: string[];
+
+  @doc("Details about performance issue. Applicable for performance impacts.")
+  @extension("x-ms-identifiers", [])
+  performance?: Performance[];
+
+  @doc("Details about connectivity issue. Applicable when root resource causing the issue is not identified. For example, when a VM is impacted due to a network issue, the impacted resource is identified as the VM, but the root cause is the network. In such cases, the connectivity field will have the details about the network issue")
+  connectivity?: Connectivity;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/no-empty-model" "Existing use case"
+  @doc("Additional fields related to impact, applicable fields per resource type are list under /impactCategories API")
+  additionalProperties?: {};
```

Comment:
how are you going to version those? can it reasonably be modeled with discriminated types?

Reply:
The additionalProperties field is set up as a flexible object because different downstream systems require different types of information. Since we're onboarding these downstream systems gradually, we want to keep things adaptable without updating spec independently. This way, we can work closely with our customers, both internal and external, to provide the right data investigating systems need. This helps us understand and investigate any issues that come up, making sure we meet everyone's needs effectively.

Reply:
@suhasaraos , here is the link to PR from where Additionproperties is modeled as object
https://github.com/Azure/azure-rest-api-specs-pr/pull/10830/files#diff-cc9f65ebaa1ef5e831fd5098958d08ce142d77632072b0f9de76ec9805f991c3

Reply:
this is already in use, refer this manifest

https://armmanifest.visualstudio.com/One/_git/Manifest?path=/MICROSOFT.IMPACT/PROD/MANIFEST.JSON&version=GBmaster&line=106&lineEnd=107&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents